web-ui/src/components/certifications/EarnedCertificationsTable.jsx

```diff
@@ -0,0 +1,566 @@
+        </DialogContent>
+        <DialogActions>
+          <Button onClick={cancelEarnedCertification}>Cancel</Button>
+          <Button
+            disabled={!selectedProfile || !selectedCertification}
+            onClick={saveEarnedCertification}
+          >
+            Save
+          </Button>
+        </DialogActions>
+      </Dialog>
+    ),
+    [earnedDialogOpen, selectedCertification, selectedEarned, selectedProfile]
+  );
+
+  const tableColumnsToUse = onlyMe ? tableColumns.slice(1) : tableColumns;
+
+  const earnedTable = useCallback(
+    () => (
+      <Card>
+        <CardHeader
+          avatar={<EmojiEvents />}
+          title="Earned Certifications"
+          titleTypographyProps={{ variant: 'h5', component: 'h2' }}
+        />
+        <CardContent>
+          <div className="row">
+            <table>
+              <thead>
+                <tr>
+                  {tableColumnsToUse.map(column => (
+                    <th
+                      key={column}
+                      onClick={() => sortTable(column)}
+                      style={{ cursor: 'pointer' }}
+                    >
+                      {column}
+                      {sortIndicator(column)}
+                    </th>
+                  ))}
+                  <th key="Actions">Actions</th>
+                </tr>
+              </thead>
+              <tbody>{earnedCertifications.map(earnedCertificationRow)}</tbody>
+            </table>
+            <IconButton
+              aria-label="Add Earned Certification"
+              classes={{ root: 'add-button' }}
+              onClick={addEarnedCertification}
+            >
+              <AddCircleOutline />
+            </IconButton>
+          </div>
+        </CardContent>
+      </Card>
+    ),
+    [earnedCertifications, sortAscending, sortColumn]
+  );
+
+  const earnedValue = useCallback(
+    earned => {
+      const certification = certificationMap[earned.certificationId];
+      switch (sortColumn) {
+        case 'Earned On':
+          return earned.earnedDate;
+        case 'Expiration':
+          return earned.expirationDate || '';
+        case 'Description':
+          return earned.description;
+        case 'Certificate Image':
+          return earned.certificateImageUrl || '';
+        case 'Badge':
+          return certification.badgeUrl || '';
+        case 'Member':
+          const profile = profileMap[earned.memberId];
+          return profile?.name ?? '';
+        case 'Name':
+          return certificationMap[earned.certificationId]?.name ?? '';
+      }
+    },
+    [certificationMap, profileMap, sortColumn]
+  );
+
+  const editEarnedCertification = useCallback(
+    earned => {
+      setSelectedEarned(earned);
+      setSelectedProfile(profileMap[earned.memberId]);
+      setSelectedCertification(certificationMap[earned.certificationId]);
+      setEarnedDialogOpen(true);
+    },
+    [certificationMap, profileMap]
+  );
+
+  const imageDialog = useCallback(
```

Comment:
When a thumbnail of a certificate image is clicked, this dialog opens to show a larger view of it.